sql/expression/function/locks.go

```diff
@@ -0,0 +1,223 @@
+func ReleaseLockFunc(ctx *sql.Context, ls *sql.LockSubsystem, lockName string) (interface{}, error) {
+	err := ls.Unlock(ctx, lockName)
+
+	if err != nil {
+		if sql.ErrLockDoesNotExist.Is(err) {
+			return nil, nil
+		} else if sql.ErrLockNotOwned.Is(err) {
+			return int8(0), nil
+		}
+
+		return nil, err
+	}
+
+	return int8(1), nil
+}
+
+// IsFreeLockFunc is the function logic that is executed when the is_free_lock function is called.
+func IsFreeLockFunc(_ *sql.Context, ls *sql.LockSubsystem, lockName string) (interface{}, error) {
+	state, _ := ls.GetLockState(lockName)
+
+	switch state {
+	case sql.LockInUse:
+		return int8(0), nil
+	default: // return 1 if the lock is free.  If the lock doesn't exist yet it is free
+		return int8(1), nil
+	}
+}
+
+// IsUsedLockFunc is the function logic that is executed when the is_used_lock function is called.
+func IsUsedLockFunc(ctx *sql.Context, ls *sql.LockSubsystem, lockName string) (interface{}, error) {
+	state, owner := ls.GetLockState(lockName)
+
+	switch state {
+	case sql.LockInUse:
+		return owner, nil
+	default:
+		return nil, nil
+	}
+}
+
+// GetLock is a SQL function implementing get_lock
+type GetLock struct {
+	expression.BinaryExpression
+	ls *sql.LockSubsystem
+}
+
+// CreateNewGetLock returns a new GetLock object
+func CreateNewGetLock(ls *sql.LockSubsystem) func(e1, e2 sql.Expression) sql.Expression {
+	return func(e1, e2 sql.Expression) sql.Expression {
+		return &GetLock{expression.BinaryExpression{e1, e2}, ls}
+	}
+}
+
+// Eval implements the Expression interface.
+func (gl *GetLock) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	if gl.Left == nil {
+		return nil, nil
+	}
+
+	val0, err := gl.Left.Eval(ctx, row)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if val0 == nil {
+		return nil, nil
+	}
+
+	if gl.Right == nil {
+		return nil, nil
+	}
+
+	val1, err := gl.Right.Eval(ctx, row)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if val1 == nil {
+		return nil, nil
+	}
+
+	lockName, ok := val0.(string)
+
+	if !ok {
+		return nil, ErrIllegalLockNameArgType.New(gl.Left.Type().String(), "get_lock")
+	}
+
+	timeout, err := sql.Int64.Convert(val1)
+
+	if err != nil {
+		return nil, fmt.Errorf("illegal value for timeeout %v", timeout)
+	}
+
+	err = gl.ls.Lock(ctx, lockName, time.Second * time.Duration(timeout.(int64)))
+
+	if err != nil {
+		if sql.ErrLockTimeout.Is(err) {
+			return int8(0), nil
+		}
+
+		return nil, err
+	}
+
+	return int8(1), nil
+}
+
+// String implements the Stringer interface.
+func (gl *GetLock) String() string {
+	return fmt.Sprintf("get_lock(%s, %s)", gl.Left.String(), gl.Right.String())
+}
+
+// IsNullable implements the Expression interface.
+func (gl *GetLock) IsNullable() bool {
+	return false
+}
+
+// WithChildren implements the Expression interface.
+func (gl *GetLock) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != 1 {
```

Comment:
2, not 1.

Should be an engine test that fails from this mistake

Reply:
added test coverage for this in function_test and added function tests.